docs: Generate documentation from alerting rules

gdemonet opened the pull request:
**Component**: docs, alert-tree

<!-- E.g. 'salt', 'containers', 'kubernetes', 'build', 'tests'... -->

**Context**: The deployed alerting rules should be documented in a way that allows readers to better grasp their meaning and implications.

**Summary**: Previous table gets removed, and a dedicated page is added to list all available alerts. The distinction between "simple" and "composite" rules is clearly shown.

gdemonet committed:
docs: Generate documentation from alerting rules

To better explain the alerting rules deployed by MetalK8s, we introduce
a new dedicated page for it in the Operation guide, and display the
tree structure encoded with "composite" rules.

gdemonet committed:
alert-tree: Remove non-existing alerts

`etcdMembersDown` does not exist anymore
`AlertmanagerFailedReload` only exists in the CRITICAL variant

These missing alerts were detected when rendering the generated docs.
A future improvement would add a test for checking that all referenced
children actually exist.

gdemonet committed:
docs: Remove previous list of alerting rules

This table is now obsolete, since we introduced a more detailed page
for all alert rules.
We also remove the CSV from the rule_extractor tooling, since it's not
needed anymore.

bert-e commented:
# Hello gdemonet,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/metalk8s/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @chengyanjin

* @lucieleonard

* @jbwatenbergscality

* @alexandre-allard-scality

* @gdemonet

* @teddyandrieux

* @nicolast

* @thomasdanan

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @chengyanjin

* @lucieleonard

* @jbwatenbergscality

* @alexandre-allard-scality

* @gdemonet

* @teddyandrieux

* @nicolast

* @thomasdanan










*The following options are set:* **approve**

gdemonet committed:
docs: Adjust styles for field lists

Match the theme colors, and correct the word break rules to avoid broken
display on Chrome. Also prevent arbitrary hyphens, since it would be
confusing to read among the already existing dashes in label values.

gdemonet committed:
docs: Add PromQL lexer for rule definitions

This will render the highlighting useful. Note however that colons in
metric names aren't yet supported by this lexer, even if they are
allowed by the spec. In such cases, we disable the highlighting
entirely.

gdemonet commented:
/approve

bert-e commented:
# Build failed



The build for [commit](https://github.com/scality/metalk8s/commit/04417671c7abc362f279772be983b1f9fccaf84d) did not succeed in branch improvement/generate-docs-for-alert-tree.




*The following options are set:* **approve**

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/123.0`



The following branches will **NOT** be impacted:

* `development/2.0`
* `development/2.1`
* `development/2.10`
* `development/2.11`
* `development/2.2`
* `development/2.3`
* `development/2.4`
* `development/2.5`
* `development/2.6`
* `development/2.7`
* `development/2.8`
* `development/2.9`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/123.0`



The following branches have **NOT** changed:

* `development/2.0`
* `development/2.1`
* `development/2.10`
* `development/2.11`
* `development/2.2`
* `development/2.3`
* `development/2.4`
* `development/2.5`
* `development/2.6`
* `development/2.7`
* `development/2.8`
* `development/2.9`



Please check the status of the associated issue None.

Goodbye gdemonet.
